.github/workflows/build.yaml

```diff
@@ -30,3 +30,23 @@ jobs:
     - uses: actions/checkout@v2
     - name: lint
       run: sudo ./scripts/test-in-lxd.sh ${{ matrix.image }} "make lint"
+
+  static-typing:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install mypy and typeshed
+        run: sudo apt-get install -y python3-mypy python3-typeshed
+      - name: Checkout the source branch
+        uses: actions/checkout@v3
+      - name: Run mypy on source branch
+        run: python3 -m mypy --ignore-missing-imports --check-untyped-defs subiquity subiquitycore system_setup console_conf scripts/replay-curtin-log.py | tee /tmp/mypy-source.out
+      - name: Checkout the target/base branch
```

Comment:
I think it makes more sense to checkout the common ancestor of {source, target} rather than target itself.